packages/client/src/ActionsAndReducers/dbWargames/wargames_ActionCreators.js

```diff
@@ -337,10 +337,8 @@ export const deleteSelectedAsset = (data) => {
 
 export const deleteSelectedRole = (dbName, data) => {
   return async (dispatch) => {
-    await data.roles.splice(data.key, 1)
-    const wargame = await wargamesApi.deleteRolesParticipations(dbName, data.roles)
-    data.handleChange(data.roles)
-    dispatch(setCurrentWargame(wargame))
+    const wargame = await wargamesApi.deleteRolesParticipations(dbName, data.roles, data.key)
+    _.isArray(wargame) ? await data.handleChange(wargame) : dispatch(setCurrentWargame(wargame))
     dispatch(addNotification('Role deleted.', 'warning'))

```

Comment:
@lilitkarapetyan - I'm trying to tidy the logic here.  The above `deleteRolesParticipations` returns two different types of result, which gets handled in two different ways.  (`handleChange` and `setCurrentWargame`).  Can you explain why there are to different paths followed?

Reply:
HandleChange for real-time update roles, setCurrentWargame for live update wargame. They are needed to work properly.

Reply:
i think I need to understand this a bit more.  We only call `deleteSelectedRole` from one place. What are the two separate paths? 

I've rewritten `deleteRolesParticipations`, but it is always returning the updated wargame.